src/nih_wayfinding/app/scripts/profiles/profile-service.js

```diff
@@ -0,0 +1,56 @@
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function ProfileService (ProfileProvider) {
+
+        // Private variables
+        var profilesList = [];
+
+        // Public Interface
+        var module = {
+            getProfile: getProfile,
+            getProfileNames: getProfileNames,
+            createProfile: createProfile
+        };
+
+        return module;
+
+        /**
+         * @returns Profile for given username
```

Comment:
Hmm this is an open question, on where to put the docstrings? I put them where the public interface is defined so that definition/documentation is in the same place, but this way has some advantages too.